ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -816,13 +816,19 @@ def _get_connections_core(
             instrument_type_id, self._reserved_instrument_type_ids
         )
 
+        requested_pin_or_relay_names = _to_ordered_set(
+            self._to_resolved_pin_or_relay_names(requested_pins)
```

Comment:
`requested_pins` is only used here. Can we move this construction right below it to improve readability?

Reply:
The logics were grouped in such a way that converting the pins, sites, and instruments to an iterable is done within the same block. Moving this above might compromise readability. Therefore, I would prefer to stick with the current arrangement.